### ISSUE TITLE
Mouse cursor offset after turning on a HiDPI monitor

### ISSUE THREAD

**Issue by timothyqiu** [godotengine/godot#41849]
**Godot version:** 3.2.3 beta 1 ~ rc 5

**OS/device including version:**
MacBook Pro (Retina, 13-inch, Early 2015)
Intel Iris Graphics 6100 1536 MB

**Issue description:**

With the HiDPI monitor turned off (closed the MBP lid), I opened the Godot editor using only the non-HiDPI external monitor.

After turning on the HiDPI monitor (opening the lid), the appearance of Godot editor stays normal, but the mouse cursor seems to be using a HiDPI coordinate.

Also, after turning on the HiDPI monitor, drag the editor to the HiDPI monitor, and then drag it back, the seamless display scaling fails.

**Steps to reproduce:**
1. Use only the non-HiDPI monitor, turn off the HiDPI monitor
2. Open the Godot Editor
3. Turn on the HiDPI monitor

**Minimal reproduction project:** Not project related

**Comment by bruvzg** [comment]
Can't reproduce, turning HiDPI monitor on (or switching monitor to HiDPI mode) do affect editor scale (without DPI independent mode implemented in the renderer, unlikely this can be fixed), but mouse input is handled correctly. Maybe internal monitor is not sending display reconfiguration events for some reason.

**Comment by timothyqiu** [comment]
It's strange that I can't reproduce the mouse cursor offset now either.

All I did is changing the display arrangement (move the HiDPI monitor from the left to the bottom of the LoDPI one). But I can't reproduce after I move it back.

**Comment by timothyqiu** [comment]
Nevermind... I opened the 3.2.2 version of Godot after changing the arrangement. I can still reproduce it in RC 5.

Sorry for the misinformation.

**Comment by timothyqiu** [comment]
I added a `NSLog` to the `displays_arrangement_changed` function, and it was called after turning on the HiDPI monitor.

Just to make sure, is the LoDPI monitor your primary display?

If I set the HiDPI monitor to be the primary display, the mouse input issue does not exist because the window is automatically moved to the HiDPI monitor after the monitor is turned on. My primary display is the LoDPI one, so the editor window stays on the LoDPI monitor, thus the mouse input offset.
